### ISSUE TITLE
Long wait

### ISSUE THREAD

**Issue by razzoli** [gammu/gammu#246]
Log: 
```
Tue 2016/09/27 20:22:12 gammu-smsd[3680]: gammu: Checking line: OK
Tue 2016/09/27 20:22:12 gammu-smsd[3680]: gammu: AT reply state: 1
Tue 2016/09/27 20:22:12 gammu-smsd[3680]: gammu: RECEIVED frame type 0x00/length 0x40/64
Tue 2016/09/27 20:22:12 gammu-smsd[3680]: gammu: 2B+|43C|4DM|54T|49I|3A:|20 |22"|53S|4DM|22"|2C,|311|0D |0A |41A +CMTI: "SM",1..A
Tue 2016/09/27 20:22:12 gammu-smsd[3680]: gammu: 54T|2B+|43C|50P|4DM|53S|3D=|22"|53S|4DM|22"|2C,|22"|53S|4DM|22" T+CPMS="SM","SM"
Tue 2016/09/27 20:22:12 gammu-smsd[3680]: gammu: 0D |0D |0A |2B+|43C|50P|4DM|53S|3A:|20 |311|2C,|322|300|2C,|311 ...+CPMS: 1,20,1
Tue 2016/09/27 20:22:12 gammu-smsd[3680]: gammu: 2C,|322|300|2C,|311|2C,|322|300|0D |0A |0D |0A |4FO|4BK|0D |0A  ,20,1,20....OK..
**Tue 2016/09/27 20:22:12 gammu-smsd[3680]: gammu: Incoming SMS
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: [Retrying 1 type 0x00]**
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: SENDING frame type 0x00/length 0x12/18
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 41A|54T|2B+|43C|50P|4DM|53S|3D=|22"|53S|4DM|22"|2C,|22"|53S|4DM AT+CPMS="SM","SM
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 22"|0D                                                          ".
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 1 "AT+CPMS="SM","SM""
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 2 "+CPMS: 1,20,1,20,1,20"
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 3 "OK"
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: Checking line: OK
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: AT reply state: 1
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: RECEIVED frame type 0x00/length 0x31/49
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 41A|54T|2B+|43C|50P|4DM|53S|3D=|22"|53S|4DM|22"|2C,|22"|53S|4DM AT+CPMS="SM","SM
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 22"|0D |0D |0A |2B+|43C|50P|4DM|53S|3A:|20 |311|2C,|322|300|2C, "...+CPMS: 1,20,
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 311|2C,|322|300|2C,|311|2C,|322|300|0D |0A |0D |0A |4FO|4BK|0D  1,20,1,20....OK.
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 0A                                                              .
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: SMS status received
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: Parsing +CPMS: 1,20,1,20,1,20 with +CPMS: @i, @i, @0
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: Parsed int 1
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: Parsed int 20
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: Used : 1
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: Size : 20
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: Getting phone SMS status
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: SENDING frame type 0x00/length 0x12/18
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 41A|54T|2B+|43C|50P|4DM|53S|3D=|22"|4DM|45E|22"|2C,|22"|4DM|45E AT+CPMS="ME","ME
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 22"|0D                                                          ".
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 1 "AT+CPMS="ME","ME""
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 2 "+CPMS: 0,10,0,10,1,20"
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 3 "OK"
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: Checking line: OK
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: AT reply state: 1
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: RECEIVED frame type 0x00/length 0x31/49
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 41A|54T|2B+|43C|50P|4DM|53S|3D=|22"|4DM|45E|22"|2C,|22"|4DM|45E AT+CPMS="ME","ME
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 22"|0D |0D |0A |2B+|43C|50P|4DM|53S|3A:|20 |300|2C,|311|300|2C, "...+CPMS: 0,10,
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 300|2C,|311|300|2C,|311|2C,|322|300|0D |0A |0D |0A |4FO|4BK|0D  0,10,1,20....OK.
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: 0A                                                              .
Tue 2016/09/27 20:25:31 gammu-smsd[3680]: gammu: SMS status received
```

Config:

```
[gammu]
device = /dev/ttyVIZ0
connection = at

[smsd]
service = sql
driver = native_mysql
debuglevel = 0
logfile = /var/log/gammu.log
phoneid = phone
user = root
password =
pc = localhost
database = smsd
```

**Comment by nijel** [comment]
What Gammu version is that? There has been recently some changes to timeouts (though I don't recall at which version right now).

**Comment by razzoli** [comment]
[Gammu version 1.37.0]

**Comment by razzoli** [comment]
Package: gammu
Version: 1.37.0-1build1
Priority: optional
Section: universe/comm
Origin: Ubuntu

**Comment by nijel** [comment]
Ah, now I see the problem in the log - the unsolicited code consumed data for regular modem response as well. That is same problem as observed in #239, however I somehow thought it's Windows specific problem there...

**Comment by nijel** [comment]
I've fixed issue seen in #239 and I think it does fix problem here as well. However I can't say that for sure without complete log or without knowing which action you did perform with Gammu. Can you please provide this information? (Or test current git code, however that will be probably tricky here as the modem needs to receive message in right timing to trigger this)

**Comment by razzoli** [comment]
Yes, It works fine, thx.
